### PR TITLE
Fix issue with react-hot-loader webpack config

### LINKED ISSUE #16: BUG: Update demo webpack hot config (`demo/webpack.config.hot.js`)
In https://github.com/FormidableLabs/builder-react-component/blob/master/config/webpack/demo/webpack.config.hot.js#L9 we need something like:

``` js
// Update loaders array.
mod.loaders[0].loaders = [require.resolve("react-hot-loader")]
  .concat(mod.loaders[0].loader ? [mod.loaders[0].loader] : [])
  .concat(mod.loaders[0].loaders || []);

// Remove single loader if any.
mod.loaders[0].loader = null;
```

/cc @coopy @boygirl

### PR DESCRIPTION
**Pull request by coopy**
Fixes #16.

/cc @ryan-roemer

### CODE REVIEW

**Comment by ryan-roemer** [comment]
LGTM!

File: config/webpack/demo/webpack.config.hot.js [hunk 1]
```diff
@@ -3,12 +3,16 @@
...
-  require.resolve("react-hot-loader")
-].concat(mod.loaders[0].loaders);
+
+// Update loaders array. First loader needs react-hot-loader.
+mod.loaders[0].loaders = [require.resolve("react-hot-loader")]
+  .concat(mod.loaders[0].loader ? [mod.loaders[0].loader] : [])
+  .concat(mod.loaders[0].loaders || []);
+
+// Remove single loader if any.
+mod.loaders[0].loader = null;
```
Comment (exogen) [inline config/webpack/demo/webpack.config.hot.js]: In any non-config JS file I'd expect `var firstLoader = mod.loaders[0]` to make all this a lot easier to read.

Comment (coopy) [inline config/webpack/demo/webpack.config.hot.js]: You've got it!

**Comment by exogen** [comment]
Awesome, thanks @coopy! :+1:

**Pull request merged**